test/testdata/rewriter/prop_tuple.rb

```diff
@@ -0,0 +1,26 @@
+# typed: true
+extend T::Sig
+
+class StructWithTuples < T::Struct
+  prop :unary_tuple, [Symbol]
+  prop :tuple, [Integer, String]
+  prop :triple, [Float, TrueClass, FalseClass]
+  prop :array_of_tuple, T::Array[[Integer, String]]
+  prop :tuple_with_combinators, [T.nilable(String), T.class_of(Integer)]
+  prop :combinator_with_tuple, T.nilable([Symbol, Float])
```

Comment:
OOC, do we already have tests for what happens if the types for `prop` declarations are not syntactically valid? (e.g., `prop :some_tuple, [42, 42, 42]`)

Reply:
Added below.